examples/network/exampleApplications/neighbourhoodHighlight.html

```diff
@@ -99,16 +98,20 @@
         }
       }
       var connectedNodes = network.getConnectedNodes(selectedNode);
-      var allConnectedNodes = [];
+      var allConnectedNodes = connectedNodes.slice(0);
+      var allConnectedNodesTemp = connectedNodes.slice(0);
 
-      // get the second degree nodes
+      // get the n degree nodes
       for (i = 1; i < degrees; i++) {
-        for (j = 0; j < connectedNodes.length; j++) {
-          allConnectedNodes = allConnectedNodes.concat(network.getConnectedNodes(connectedNodes[j]));
+        for (j = 0; j < allConnectedNodes.length; j++) {
+          allConnectedNodesTemp = allConnectedNodesTemp.concat(network.getConnectedNodes(allConnectedNodesTemp[j]));
+
         }
+        allConnectedNodes = allConnectedNodesTemp.filter(function(item, i, ar){ return ar.indexOf(item) === i; }); //Remove duplication
```

Comment:
After thinking hard, I can't think of a better way to get the unique values than this one. I find it wasteful to have to copy the nodes array, but this is the best way to ensure unique values.

_(no action required)_